openfe/protocols/openmm_utils/system_creation.py

```diff
@@ -235,3 +245,217 @@ def _add_small_mol(comp,
+    ----------
+    offmol: openff.toolkit.Molecule
+      The Molecule to assign AM1BCC charges for.
+    nagl_model : Optional[str]
+      The nagl model to use for partial charge assignment.
+      If None, will use latest available model.
+    """
+
+    if nagl_model is None:
+        # It's not fully clear that the models will always be sort ordered
+        # see: https://github.com/openforcefield/openff-nagl-models/issues/12
+        from openff.nagl_models import list_available_nagl_models
+        nagl_model = list_available_models()[-1]
+
+    offmol.assign_partial_charges(
+        partial_charge_method=nagl_model,
+        toolkit_registry=NAGLToolkitWrapper(),
+    )
+
+
+def assign_am1bccelf10_charges(
+    offmol: OFFMol,
+    charge_backend: Optional[str] = 'ambertools',
+    ambertools_generate_n_conformers: int = 500,
+) -> None:
+    """
+    Assign AM1BCC charges using a given toolkit.
+
+    Parameters
+    ----------
+    offmol : openff.toolkit.Molecule
+      The Molecule to assign AM1BCCELF10 charges for.
+    charge_backend : str
+      Which backend to use to generate charges. Available options are:
+
+      * `ambertools`: Use AmberTools' antechamber to generate charges.
+      * `oechem`: Use the OpenEye toolkit to generate charges.
+
+    ambertools_generate_n_conformers : int
+      The number of conformers to initially generate for elf10 selection.
+      Note: this is only used by the `ambertools` backend option.
+    """
+    backend = _get_toolkit_wrapper_charge_backend(charge_backend)
+
+    if charge_backend.lower() == 'oechem':
+        offmol.assign_partial_charges(
+            partial_charge_method='am1bccelf10',
+            toolkit_registry=backend(),
+        )
+
+    else:
+        # make a copy of the offmol to avoid overwriting conformers
+        # TODO: in tests ensure that the offmol conformer doesn't change
+        # after charge asisgnment for any of these methods!
+        offmol_copy = OFFMol(offmol)
+
+        # We generate conformers using RDKit to be consistent
+        offmol_copy.generate_conformers(
+            # The OFF recommended default for OpenEye is 500
+            n_conformers=ambertools_generate_n_conformers,
+            rms_cutoff=0.25 * offunit.angstrom,
+            toolkit_registry=RDKitToolkitWrapper()
+        )
+
+        # Next we apply the elf10 conformer selection
+        offmol_copy.apply_elf_conformer_selection(
+            percentage=2,
+            limit=10,
+            rms_tolerance=0.05 * offunit.angstrom,  # Should this be an option?
+        )
+
+        # Then we loop over the selected conformers
+        charges = np.zeros([offmol_copy.n_atoms])
+
+        for conf in offmol_copy.conformers:
+            offmol_copy.assign_partial_charges('am1bcc', use_conformers=[conf])
+            charges += offmol_copy.partial_charges
+
+        charges /= len(offmol_copy.conformers)
+
+        offmol.partial_charges = charges
+
+
+def assign_partial_charges(
```

Comment:
I first was a bit confused about this function since it has the same name as the openff toolkit function, but maybe that is intentional?

Reply:
This was mostly a "well I guess this method name succintly describes what it does quite well" rather than an intentional name duplication.

I don't really have any attachements tbh.